com.trekglobal.idempiere.rest.api/src/com/trekglobal/idempiere/rest/api/json/LocationTypeConverter.java

```diff
@@ -0,0 +1,349 @@
+* Contributors:                                                       *
+* - Trek Global Corporation                                           *
+* - Heng Sin Low                                                      *
+**********************************************************************/
+package com.trekglobal.idempiere.rest.api.json;
+
+import org.compiere.model.GridField;
+import org.compiere.model.Lookup;
+import org.compiere.model.MCity;
+import org.compiere.model.MColumn;
+import org.compiere.model.MCountry;
+import org.compiere.model.MLocation;
+import org.compiere.model.MLocationLookup;
+import org.compiere.model.MProcessPara;
+import org.compiere.model.MRegion;
+import org.compiere.util.Env;
+import org.compiere.util.Msg;
+import org.compiere.util.Util;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+/**
+ * json type converter for C_Location
+ * @author muriloht (devCoffee, muriloht@devcoffee.com.br)
+ *
+ */
+public class LocationTypeConverter implements ITypeConverter<Object> {
+
+	/**
+	 * default constructor
+	 */
+	public LocationTypeConverter() {
+		
+	}
+
+	@Override
+	public Object toJsonValue(MColumn column, Object value) {
+		String label = Msg.getElement(Env.getCtx(), column.getColumnName());
+		Lookup lookup = new MLocationLookup(Env.getCtx(), 0);
+		return toJsonValue(column.getAD_Reference_ID(), label, lookup, column.getReferenceTableName(), value);
+	}
+
+	@Override
+	public Object toJsonValue(GridField field, Object value) {
+		return toJsonValue(field.getDisplayType(), field.getHeader(), field.getLookup(), getReferenceTableNameFromField(field), value);
+	}
+	
+	private String getReferenceTableNameFromField(GridField field) {
+		String refTableName = null;
+		if (field.getVO().isProcess) {
+			MProcessPara pp = field.getAD_Column_ID() > 0 ? MProcessPara.get(field.getAD_Column_ID()) : null;
+			if (pp != null)
+				refTableName = pp.getReferenceTableName();
+		} else {
+			MColumn column = field.getAD_Column_ID() > 0 ? MColumn.get(Env.getCtx(), field.getAD_Column_ID()) : null;
+			if (column != null)
+				refTableName = column.getReferenceTableName();
+		}
+		return refTableName;
+	}
+	
+	private Object toJsonValue(int displayType, String label, Lookup lookup, String refTableName, Object value) {
+		if (lookup != null) {
+			
+			MLocation loc = MLocation.get((Integer)value);
+			
+			JsonObject ref = new JsonObject();
+			ref.addProperty("propertyLabel", label);
+			if (value instanceof Number)
+				ref.addProperty("id", ((Number)value).intValue());
+			else
+				ref.addProperty("id", value.toString());
+			String display = lookup.getDisplay(value);
+			if (!Util.isEmpty(display, true)) {
+				ref.addProperty("identifier", display);
+			}							
+			if (!Util.isEmpty(refTableName))
+				ref.addProperty("model-name", refTableName.toLowerCase());
+			
+			if (!Util.isEmpty(loc.getAddress1()))
```

Comment:
Hi @muriloht 

It would be better if we iterate here in the columns of C_Location.

I'm thinking about custom columns in C_Location are not added here to the json response.

So, my proposal would be:
- iterate in the columns
- add special treatment for the columns hardcoded here, like C_Country_ID, C_Region_ID, etc
- the rest of columns can follow the approach for Address1..5 - if not empty, add it

WDYT?